src/backend/columnar/columnar_tableam.c

```diff
@@ -1910,6 +1910,11 @@ ColumnarSubXactCallback(SubXactEvent event, SubTransactionId mySubid,
 void
 columnar_tableam_init()
 {
+	ColumnarTableSetOptions_hook_type **ColumnarTableSetOptions_hook_ptr =
+		(ColumnarTableSetOptions_hook_type **) find_rendezvous_variable(
+			"ColumnarTableSetOptions_hook");
+	*ColumnarTableSetOptions_hook_ptr = &ColumnarTableSetOptions_hook;
```

Comment:
Currently we rely on this code being executed before we populate the hook we want to call.

If we would not store `&ColumnarTableSetOptions_hook` in the rendezvous variable, but instead directly use the rendezvous variable when invoking the hook we can remove the reliance on code execution order, incase that might change in the future.

eg. instead of having `ColumnarTableSetOptions_hook` be a function pointer we can store the rendezvous variable there. When we dispatch we can dereference twice (with appropriate NULL checks before the second dereference).

Although it causes a minimal overhead when calling the hook, it feels more future proof to support arbitrary loading order.

Given changing columnar options is not performance critical I'd rather refactor this slightly to not have 'races' influence the behaviour.

Reply:
The way it's done currently matches the expected convention for a rendezvous variable -- pointing to a variable rather than using the hash table entry itself. Also, if there's something weird about the loading order, we'd find out right faster rather than waiting until a columnar option changes.

Because citus's init calls `load_file()` directly, it effectively specifies the load order, so even now, it's not dependent on `shared_preload_libraries`. I think it's OK this way.

Also, we probably will need columnar to load first in the long term anyway, because citus is dependent on columnar. I think we'll have that `load_file` near the top of citus's init indefinitely.

Reply:
On second thought, I did use a variable to hold the symbol. Little downside, perhaps clarifies.